[BUG] Processing some mail runs in to error

### Description

Some mail processing runs into an error and the mail attachments are not imported. It could have to do with the change from #4931 because I have not had problems at the beginning of december.

### Steps to reproduce

1. Account on web.de
2. Activate the Deutsche Post Briefankündigung
3. Setup E-Mail processing
4. Get a Mail send per E-Post to you containing a digital copy of the letter as PDF and an image (png) of the envelop.

### Webserver logs

```bash
[2023-12-23 03:40:01,814] [ERROR] [paperless_mail] Rule Web.de.ePost Digitale Kopie: Error while processing mail 159: [Errno 21] Is a directory: '/tmp/paperless/paperless-mail-9ixp8o7a'
Traceback (most recent call last):
  File "/usr/src/paperless/src/paperless_mail/mail.py", line 591, in _handle_mail_rule
    processed_files = self._handle_message(message, rule)
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/paperless_mail/mail.py", line 639, in _handle_message
    processed_elements += self._process_attachments(
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/paperless/src/paperless_mail/mail.py", line 715, in _process_attachments
    temp_filename.write_bytes(att.payload)
  File "/usr/local/lib/python3.11/pathlib.py", line 1067, in write_bytes
    with self.open(mode='wb') as f:
         ^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/pathlib.py", line 1044, in open
    return io.open(self, mode, buffering, encoding, errors, newline)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
IsADirectoryError: [Errno 21] Is a directory: '/tmp/paperless/paperless-mail-9ixp8o7a'
```


### Browser logs

_No response_

### Paperless-ngx version

2.1.3

### Host OS

Ubuntu 22.04

### Installation method

Docker - official image

### Browser

_No response_

### Configuration changes

_No response_

### Other

_No response_

### Please confirm the following

- [X] I believe this issue is a bug that affects all users of Paperless-ngx, not something specific to my installation.
- [X] I have already searched for relevant existing issues and discussions before opening this report.
- [X] I have updated the title field above with a concise description.

--- a/src/paperless_mail/mail.py
+++ b/src/paperless_mail/mail.py
@@ -703,6 +703,12 @@ class MailAccountHandler(LoggingMixin):
             mime_type = magic.from_buffer(att.payload, mime=True)
 
             if is_mime_type_supported(mime_type):
+                self.log.info(
+                    f"Rule {rule}: "
+                    f"Consuming attachment {att.filename} from mail "
+                    f"{message.subject} from {message.from_}",
+                )
+
                 os.makedirs(settings.SCRATCH_DIR, exist_ok=True)
 
                 temp_dir = Path(
@@ -711,14 +717,15 @@ class MailAccountHandler(LoggingMixin):
                         dir=settings.SCRATCH_DIR,
                     ),
                 )
-                temp_filename = temp_dir / pathvalidate.sanitize_filename(att.filename)
-                temp_filename.write_bytes(att.payload)
 
-                self.log.info(
-                    f"Rule {rule}: "
-                    f"Consuming attachment {att.filename} from mail "
-                    f"{message.subject} from {message.from_}",
-                )
+                attachment_name = pathvalidate.sanitize_filename(att.filename)
+                if attachment_name:
+                    temp_filename = temp_dir / attachment_name
+                else:  # pragma: no cover
+                    # Some cases may have no name (generally inline)
+                    temp_filename = temp_dir / "no-name-attachment"
+
+                temp_filename.write_bytes(att.payload)
 
                 input_doc = ConsumableDocument(
                     source=DocumentSource.MailFetch,

